standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/DirectSqlUpdatePart.java

```diff
@@ -526,18 +519,18 @@ public Map<String, Map<String, String>> updatePartitionColumnStatistics(Map<Stri
         MetaStoreListenerNotifier.notifyEventWithDirectSql(transactionalListeners,
                 EventMessage.EventType.UPDATE_PARTITION_COLUMN_STAT_BATCH, eventBatch, dbConn, sqlGenerator);
       }
-      dbConn.commit();
-      committed = true;
+      closeDbConn(jdoConn);
```

Comment:
this won't close the underlying connection because the current active is active, see 
https://github.com/datanucleus/datanucleus-core/blob/efba3bd0e8345e1171d5fe529cae693293f178ae/src/main/java/org/datanucleus/store/AbstractStoreManager.java#L378-L392
It tells we have reached the end of `transaction` and is able to commit now, until the `tx.commit()` the underlying connection will return back to the connection pool.

Reply:
If the `tx.commit()` happens before the `closeDbConn(jdoConn)`, a JDOUserException cloud see when commit the change: 
`The Connection was acquired by the developer and must be closed before using the persistence API`.

https://www.datanucleus.org/products/accessplatform_6_0/jdo/persistence.html#_user_connection
> You must return the connection back to the PersistenceManager before performing any PersistenceManager operation. You do this by calling conn.close(). If you don’t return the connection and try to perform an PersistenceManager operation which requires the connection then a JDOUserException is thrown.